test/testdata/namer/constant_in_block.rb

```diff
@@ -1,15 +1,17 @@
 # typed: true
 class A
-  
+
   def self.foo
     foo do
     C = 1 # this is parse error in Ruby, but not in Sorbet
+  # ^^^^^ error: Dynamic constant assignment
     C # error: Unable to resolve constant
     end
   end
-  
+
   [].each do
     B = 1
+  # ^^^^^ error: Dynamic constant assignment
```

Comment:
I'm on the fence about whether or not this should be an error. I can see not raising an error if the owning context doesn't involve a method definition.

Reply:
It turns out the error added in this PR doesn't get triggered here, as the owning context is the class, not the block.